locals.tf

```diff
@@ -1,53 +1,41 @@
 locals {
 
+  # EKS Cluster
+  cluster_id              = coalescelist(aws_eks_cluster.this[*].id, [""])[0]
+  cluster_arn             = coalescelist(aws_eks_cluster.this[*].arn, [""])[0]
+  cluster_name            = coalescelist(aws_eks_cluster.this[*].name, [""])[0]
+  cluster_endpoint        = coalescelist(aws_eks_cluster.this[*].endpoint, [""])[0]
+  cluster_auth_base64     = coalescelist(aws_eks_cluster.this[*].certificate_authority[0].data, [""])[0]
+  cluster_oidc_issuer_url = flatten(concat(aws_eks_cluster.this[*].identity[*].oidc[0].issuer, [""]))[0]
+
   cluster_security_group_id         = var.cluster_create_security_group ? join("", aws_security_group.cluster.*.id) : var.cluster_security_group_id
-  cluster_primary_security_group_id = var.cluster_version >= 1.14 ? element(concat(aws_eks_cluster.this[*].vpc_config[0].cluster_security_group_id, [""]), 0) : null
+  cluster_primary_security_group_id = try(var.cluster_version >= 1.14, false) ? element(concat(aws_eks_cluster.this[*].vpc_config[0].cluster_security_group_id, [""]), 0) : null
```

Comment:
1.14 can be removed at all as this is not supported anymore and there is no way to create it. 

Also AWS states:
```
Q: What happens on the end of support date?

A: On the end of support date, you can no longer create new Amazon EKS clusters with the unsupported version. Existing control planes are automatically updated by Amazon EKS to the oldest supported version through a gradual deployment process after the end of support date. After the automatic control plane update, you must manually update cluster add-ons and Amazon EC2 nodes. For more information, see To update the Kubernetes version for your Amazon EKS cluster .
```

https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html

Reply:
Good point. I will add `minimum_cluster_version` variable with the default value "1.16".

Reply:
Simplified to remove the check completely.